components/compliance-service/ingest/ingestic/ingestic.go

```diff
@@ -900,9 +895,99 @@ func (backend *ESClient) GetDocByReportUUId(ctx context.Context, data *relaxting
 				if err != nil {
 					logrus.Errorf("Received error while unmarshling %+v", err)
 				}
+
 			}
 
 		}
 	}
+
 	return &item, nil
 }
+
+func (backend *ESClient) CheckIfControlIdExistsForToday(docId string, indexToday string) (bool, error) {
+	logrus.Debugf("Checking the control document exists for today with doc Id :%s", docId)
+	fsc := elastic.NewFetchSourceContext(false)
+	boolQuery := elastic.NewBoolQuery()
+	idsQuery := elastic.NewIdsQuery()
+	idsQuery.Ids(docId)
+	boolQuery = boolQuery.Must(idsQuery)
+	searchSource := elastic.NewSearchSource().
+		FetchSourceContext(fsc).
+		Query(boolQuery).
+		Size(1)
+	searchResult, err := backend.client.Search().
+		SearchSource(searchSource).
+		Index(indexToday).
+		Do(context.Background())
+	if err != nil {
+		switch {
+		case elastic.IsTimeout(err):
+			logrus.Errorf("Timeout retrieving document: %v", err)
+			return false, err
+		default:
+			logrus.Errorf("Received error: %v", err)
+			return false, err
+		}
+	}
+
+	if searchResult.TotalHits() > 0 {
+		// Iterate through results
+		for _, hit := range searchResult.Hits.Hits {
+			// hit.Index contains the id of the index
+			if len(hit.Id) > 0 {
+				logrus.Debugf("Found the document with for control with doc Id %s", docId)
+				return true, nil
+			}
+
+		}
+	}
+	return false, nil
+
+}
+
+func (backend *ESClient) UploadDataToControlIndex(ctx context.Context, reportuuid string, controls []relaxting.Control, endTime time.Time) error {
+	mapping := mappings.ComplianceControlRepData
+	index := mapping.IndexTimeseriesFmt(endTime)
+
+	bulkRequest := backend.client.Bulk()
+	for _, control := range controls {
+		docId := GetDocIdByControlIdAndProfileID(control.ControlID, control.Profile.ProfileID)
+		found, err := backend.CheckIfControlIdExistsForToday(docId, index)
+		if err != nil {
```

Comment:
In case of error we should not check if the doc is found or not.
What should we do in case of failure? If we want to continue then add the `continue` statement. If we want to retry then we need to add the retry logic.

Reply:
done